test/malloc_compliance_tests.cpp

```diff
@@ -0,0 +1,156 @@
+// Copyright (C) 2023 Intel Corporation
+// Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+// Basic tests
+// ISO/IEC 9899:201x 7.22.3 compliance
+
+#include <errno.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "malloc_compliance_tests.hpp"
+#include "test_helpers.h"
+#include "umf/memory_pool.h"
+
+#include "base.hpp"
+using umf_test::test;
+
+//------------------------------------------------------------------------
+// Configurable defs
+//------------------------------------------------------------------------
+
+#define KB 1024
+#define MB (1024 * KB)
+
+#define MAX_ALLOC_SIZE (1 * MB)
+#define ITERATIONS 100
+
+//------------------------------------------------------------------------
+// Helpers
+//------------------------------------------------------------------------
+
+#define ASSERT_MSG(expr, msg)                                                  \
```

Comment:
in gtest I believe you can do something like `ASSERT(whatever) << "my error message"`.

Reply:
Indeed. Done.